src/util/rbs_constant_pool.c

```diff
@@ -0,0 +1,107 @@
+#include "rbs/util/rbs_constant_pool.h"
+#include "ruby.h" // Temporarily used for rb_intern2().
+#include "ruby/encoding.h" // Temporarily used for rb_usascii_encoding().
+
+
+#ifdef _MSC_VER
+    _STATIC_ASSERT(
+        sizeof(ID) == sizeof(rbs_constant_id_t)
+    );
+    _STATIC_ASSERT(
+        sizeof(VALUE) == sizeof(rbs_constant_t)
+    );
+#else
+    _Static_assert(
+        __builtin_types_compatible_p(ID, rbs_constant_id_t),
+        "rbs_constant_id_t must be the same as a Ruby ID for now."
+    );
+
+    _Static_assert(
+        __builtin_types_compatible_p(VALUE, rbs_constant_t),
+        "rbs_constant_t must be the same as a Ruby Symbol for now."
+    );
```

Comment:
These assertions confirm that our temporary definitions of `rbs_constant_id_t` and `rbs_constant_t` are memory-compatible with `ID` and `VALUE` (Symbol), so that we can safely cast back/forth between them. They'll go away once I add in the pure C implementation of the constant pool.